src/XenobladeChroniclesX/Enhancements/AntiAliasing/patch_xcxaa.asm

```diff
@@ -0,0 +1,316 @@
+lis  r3, AASharpness@ha
+lfs  f1, AASharpness@l(r3)
+addi r3, r10, 1
+lis r4, str_AA_floating_precision@ha
+ori r4, r4, str_AA_floating_precision@l
+;updated the format string with the user specified decimal precision
+li r25, 0x30+$AASharpnessDecimalPlaces
+stb r25, 2(r4)
+;sets condition register and branches to sprintf to convert the float in f1 to a string
+;at the memory address in r3, using the string format pointed to by r4
+CREQV 4*cr1+eq, 4*cr1+eq, 4*cr1+eq
+mtctr r29 ;bl sprintf
+bctrl
+
+bl _ChangeAntiAliasingSettingINIMod
+
+;AA_Range Setting ---------------------------------------------------------
+_ChangeAARangeSettingINIMod:
+bl _DeleteINIAACopy
+bl _makeCopyofINIAA
+
+lis r25, str_AA_Range@ha
+ori r25, r25, str_AA_Range@l
+bl _CheckForINIAASetting
+
+lis  r3, AARange@ha
+lfs  f1, AARange@l(r3)
+addi r3, r10, 1
+lis r4, str_AA_floating_precision@ha
+ori r4, r4, str_AA_floating_precision@l
+li r25, 0x30+$AARangeDecimalPlaces
+stb r25, 2(r4)
+CREQV 4*cr1+eq, 4*cr1+eq, 4*cr1+eq
+mtctr r29 ;bl sprintf
+bctrl
+
+bl _ChangeAntiAliasingSettingINIMod
+
+;AA_Threshold Setting ---------------------------------------------------------
+_ChangeAAThresholdSettingINIMod:
+bl _DeleteINIAACopy
+bl _makeCopyofINIAA
+
+lis r25, str_AA_Threshold@ha
+ori r25, r25, str_AA_Threshold@l
+bl _CheckForINIAASetting
+
+lis  r3, AAThreshold@ha
+lfs  f1, AAThreshold@l(r3)
+addi r3, r10, 1
+lis r4, str_AA_floating_precision@ha
+ori r4, r4, str_AA_floating_precision@l
+li r25, 0x30+$AAThresholdDecimalPlaces
+stb r25, 2(r4)
+CREQV 4*cr1+eq, 4*cr1+eq, 4*cr1+eq
+mtctr r29 ;bl sprintf
+bctrl
+
+bl _ChangeAntiAliasingSettingINIMod
+
+;AA_ThresholdMin Setting ---------------------------------------------------------
+_ChangeAAThresholdMINSettingINIMod:
+bl _DeleteINIAACopy
+bl _makeCopyofINIAA
+
+lis r25, str_AA_ThresholdMin@ha
+ori r25, r25, str_AA_ThresholdMin@l
+bl _CheckForINIAASetting
+
+lis  r3, AAThresholdMin@ha
+lfs  f1, AAThresholdMin@l(r3)
+addi r3, r10, 1
+lis r4, str_AA_floating_precision@ha
+ori r4, r4, str_AA_floating_precision@l
+li r25, 0x30+$AAThresholdMinDecimalPlaces
+stb r25, 2(r4)
+CREQV 4*cr1+eq, 4*cr1+eq, 4*cr1+eq
+mtctr r29 ;bl sprintf
+bctrl
+
+bl _ChangeAntiAliasingSettingINIMod
+
+
+;restores the game to how it was before the mod and exits the mod
+_ExitEditINIforAAmod:
+;delets the back up of the lib.ini file that we made to edit the original
+bl _DeleteINIAACopy
+;restore registers
+lwz r25, 0x1200+0x800+128(r22)
+mtlr r25
+lmw r0, 0x1200+0x800(r22)
+;delete register backup
+li r28, 0
+addi r26, r22, 0x1200+0x800-4
+li r25, (128+4)/4
+mtctr r25
+_forloopClearRegistersAA:
+ stwu r28, 4(r26)
+ bdnz _forloopClearRegistersAA
+
+;restores line we over wrote and returns to the game
+;lwz r22, 0xa0(r31)
+lis r26, 0x101c
+blr
+
+
+[XCX_AntiAliasing_V101E]
+moduleMatches = 0xF882D5CF, 0x218F6E07 ; 1.0.1E, 1.0.0E
+.origin = codecave
+_SprintfFunctionAddress_V101E_AA:
+stmw r0, 0x1200+0x800(r22) ;backs up current registers
```

Comment:
This graphic pack looks very complex, but I've got many questions about the code itself. From the looks of it, it seems to be writing to some random offset and using it as a stack to read a file into? Kinda seems like a delayed crash or corruption is bound to happen because of it. Let's maybe talk on Discord more about this graphic pack in particular.